.github/workflows/prof_asan.yml

```diff
@@ -0,0 +1,58 @@
+name: Profiling ASAN Tests
+
+on:
+  pull_request:
+
+jobs:
+  prof-correctness:
+    runs-on: ubuntu-latest
+    container:
+      image: datadog/dd-trace-ci:php-8.3_bookworm-1
+      # https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#user
+      options: --user root
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-asan-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build profiler
+        run: |
+          set -eux
+          switch-php nts-asan
+          cd profiling
+          export CARGO_TARGET_DIR=/tmp/build-cargo
+          export CC=clang-16
+          export CFLAGS='-fsanitize=address  -fno-omit-frame-pointer'
+          export LDFLAGS='-fsanitize=address'
+          export RUSTC_LINKER=lld-16
+          triplet=$(uname -m)-unknown-linux-gnu
+          RUSTFLAGS='-Zsanitizer=address' cargo +nightly-2023-05-03 build -Zbuild-std --target $triplet --release
+          cd tests
+          cp -v $(php-config --prefix)/lib/php/build/run-tests.php .
+          php run-tests.php --asan -d extension=$CARGO_TARGET_DIR/$triplet/release/libdatadog_php_profiling.so phpt
```

Comment:
This is just a minor note and not blocking: we could extract this into a "Run tests" step after the "Cache build dependencies" step, so that build dependencies will be cached even in case the actual tests fail.

Reply:
Good point 👍🏻 .